gateway/main.go

```
@@ -18,19 +20,40 @@ func main() {
 
 	ctx := signals.Context()
 
-	{
-		address, token, opts, err := apiClientConfig()
-		if err != nil {
-			log.Fatal(err)
-		}
-		scrapeInterval, err := scrapeDuration()
-		if err != nil {
-			log.Fatal(err)
+	address, token, opts, err := apiClientConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	apiClient := sdk.NewAPIClient(address, token, &opts)
+
+	noiseInterval, err := noiseFrequency()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ticker := time.NewTicker(noiseInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			createEvent(apiClient, ctx)
+		case <-ctx.Done():
+			return
 		}
-		go newNoisyNeighbor(
-			sdk.NewAPIClient(address, token, &opts),
-			time.Duration(scrapeInterval),
-		).run(ctx)
 	}
 
 }
+
+func createEvent(apiClient sdk.APIClient, ctx context.Context) {
+	_, err := apiClient.Core().Events().Create(
+		ctx,
+		core.Event{
+			Source: "github.com/willie-yao/brigade-noisy-neighbor",
+			Type:   "noise",
+		},
+	)
+	if err != nil {
+		log.Fatal(err)
```

Let's return this error. And that should make this function more testable.